packages/region/src/services/region-module.ts

```
@@ -223,37 +183,167 @@ export default class RegionModuleService<
+    // Somewhat less efficient, but region operations will be very rare, so it is better to go with a simple solution
+    await this.countryService_.update(
+      {
+        selector: {
+          region_id: normalizedInput
+            .filter((region) => !!region.countries)
+            .map((region) => region.id)
+            .flat(),
+        },
+        data: { region_id: null },
+      },
+      sharedContext
+    )
+
+    const validations = [
+      this.validateCurrencies(
+        normalizedInput.map((d) => d.currency_code),
+        sharedContext
+      ),
+      this.validateCountries(
+        normalizedInput.map((d) => d.countries ?? []).flat(),
+        sharedContext
+      ),
+    ] as const
+
+    // Assign the full country object so the ORM updates the relationship
+    const [, dbCountries] = await Promise.all(validations)
+    const dbCountriesMap = asMap(dbCountries, "iso_2")
+    let normalizedDbRegions = normalizedInput.map((region) =>
+      removeUndefined({
+        ...region,
+        countries: region.countries?.map((c) => dbCountriesMap[c]),
+      })
+    )
+
+    return await this.regionService_.update(normalizedDbRegions, sharedContext)
+  }
+
+  private normalizeInput<T extends UpdatableRegionFields>(regions: T[]): T[] {
+    return regions.map((region) =>
+      removeUndefined({
+        ...region,
+        currency_code: region.currency_code?.toLowerCase(),
+        name: region.name?.trim(),
+        countries: region.countries?.map((country) => country.toLowerCase()),
       })
+    )
+  }
 
-      result = await this.regionService_.update(toUpdate, sharedContext)
+  @InjectTransactionManager("baseRepository_")
+  private async validateCurrencies(
+    currencyCodes: (string | undefined)[] | undefined,
+    sharedContext: Context
+  ): Promise<void> {
+    const normalizedCurrencyCodes = currencyCodes
+      ?.filter((c) => c !== undefined)
```

todo: here null will go to the map step. 
```ts
currencyCodes
      ?.filter((c) => !!c)
      .map(c => c.toLowerCase()
```

---

Well actually I was trying to handle the scenario where someone passes an empty string, for example. If they do, the validation below will catch it, but if we filter it out, it won't, so checking undefined only is better.

---

And we do this because we want to throw when empty strings are passed, because that is, in fact, invalid data?

---

Yes exactly @olivermrbl